internal/cmd/pagination.go

```diff
@@ -0,0 +1,50 @@
+package cmd
+
+import (
+	"github.com/infrahq/infra/api"
+	"github.com/infrahq/infra/internal/logging"
+)
+
+// listAll is a helper function that handles pagination and calls the given list request function.
+// listItems is the corresponding function in the API client that handles the Request "req".
+// handleError is a function that handles the error returned by the API client.
+func listAll[Item any, Req api.Paginatable](client *api.Client, req Req, listItems func(api.Client, Req) (*api.ListResponse[Item], error)) ([]Item, error) {
```

Comment:
We can accept just the api client method, instead of accepting client and the function separately.

```suggestion
func listAll[Item any, Req api.Paginatable](req Req, listItems func(Req) (*api.ListResponse[Item], error)) ([]Item, error) {
```

Then call `listAll` like this:

```go
listAll(api.ListDestinationsRequest{}, client.ListDestinations)
```